tools/generate_sql_introspection.py

```diff
@@ -0,0 +1,90 @@
+import asyncpg
+import os
+import asyncio
+import json
+
+
+async def run():
+
+    localdev = os.path.expanduser('~/.local/share/edgedb/_localdev')
+
+    c = await asyncpg.connect(
+        host=localdev, database='postgres', user='postgres'
+    )
+
+    res = await c.fetch(
+        '''
+    WITH
+    columns AS (
+        SELECT
+            table_schema,
+            table_name,
+            TO_JSON(ARRAY [
+                column_name, COALESCE(domain_name, data_type)
+            ]) AS col
+        FROM information_schema.columns
+        WHERE (
+            table_schema = 'information_schema' AND table_name NOT LIKE '\_pg%'
+        ) OR (
+            table_schema = 'pg_catalog'
+        )
+        ORDER BY table_schema, table_name, ordinal_position
+    ),
+    tables AS (
+        SELECT table_schema, table_name, JSON_AGG(col) as cols
+        FROM columns
+        GROUP BY table_schema, table_name
+    ),
+    schemas AS (
+        SELECT table_schema, JSON_OBJECT_AGG(table_name, cols) as tables
+        FROM tables
+        GROUP BY table_schema
+    )
+    SELECT JSON_OBJECT_AGG(table_schema, tables)
+    FROM schemas
+    '''
+    )
+
+    schema_json = res[0][0]
+    schema = json.loads(schema_json)
+
+    pos = 0
+    with open('./edb/pgsql/resolver/sql_introspection.py', 'r') as file:
```

Comment:
I'd recommend just embedding the header of the file in this script and outputting it, instead of trying to update it in place.

Also mention in the header that the file is generated (like `edb/errors/__init__.py`)